src/sqlancer/mysql/oracle/MySQLCERTOracle.java

```diff
@@ -0,0 +1,248 @@
+import sqlancer.common.oracle.TestOracle;
+import sqlancer.common.query.SQLQueryAdapter;
+import sqlancer.common.query.SQLancerResultSet;
+import sqlancer.mysql.MySQLErrors;
+import sqlancer.mysql.MySQLGlobalState;
+import sqlancer.mysql.MySQLSchema.MySQLTables;
+import sqlancer.mysql.MySQLVisitor;
+import sqlancer.mysql.ast.MySQLBinaryLogicalOperation;
+import sqlancer.mysql.ast.MySQLBinaryLogicalOperation.MySQLBinaryLogicalOperator;
+import sqlancer.mysql.ast.MySQLColumnReference;
+import sqlancer.mysql.ast.MySQLExpression;
+import sqlancer.mysql.ast.MySQLSelect;
+import sqlancer.mysql.ast.MySQLTableReference;
+import sqlancer.mysql.gen.MySQLExpressionGenerator;
+
+public class MySQLCERTOracle extends CERTOracleBase<MySQLGlobalState> implements TestOracle<MySQLGlobalState> {
+    private MySQLExpressionGenerator gen;
+    private MySQLSelect select;
+
+    public MySQLCERTOracle(MySQLGlobalState globalState) {
+        super(globalState);
+        MySQLErrors.addExpressionErrors(errors);
+    }
+
+    @Override
+    public void check() throws SQLException {
+        queryPlan1Sequences = new ArrayList<>();
+        queryPlan2Sequences = new ArrayList<>();
+
+        // Randomly generate a query
+        MySQLTables tables = state.getSchema().getRandomTableNonEmptyTables();
+        gen = new MySQLExpressionGenerator(state).setColumns(tables.getColumns());
+        List<MySQLExpression> fetchColumns = new ArrayList<>();
+        fetchColumns.addAll(Randomly.nonEmptySubset(tables.getColumns()).stream()
+                .map(c -> new MySQLColumnReference(c, null)).collect(Collectors.toList()));
+        List<MySQLExpression> tableList = tables.getTables().stream().map(t -> new MySQLTableReference(t))
+                .collect(Collectors.toList());
+
+        select = new MySQLSelect();
+        select.setFetchColumns(fetchColumns);
+        select.setFromList(tableList);
+
+        select.setSelectType(Randomly.fromOptions(MySQLSelect.SelectType.values()));
+        if (Randomly.getBoolean()) {
+            select.setWhereClause(gen.generateExpression());
+        }
+        if (Randomly.getBoolean()) {
+            select.setGroupByExpressions(fetchColumns);
+            if (Randomly.getBoolean()) {
+                select.setHavingClause(gen.generateExpression());
+            }
+        }
+
+        // Set the join. Todo: to make it random
+        // List<MySQLExpression> joinExpressions = getJoins(tableList, state);
+        // select.setJoinList(joinExpressions);
+
+        // Get the result of the first query
+        String queryString1 = MySQLVisitor.asString(select);
+        int rowCount1 = getRow(state, queryString1, queryPlan1Sequences);
+
+        boolean increase = false;
+        // Mutate the query
+        // Disable limit due to its false positive
+        Mutator mutation = Mutator.getRandom();
+        switch (mutation) {
+        case JOIN:
+            // increase = mutateJoin();
```

Comment:
not sure if I understand the code here. In the case of a `JOIN`, you fall through to the `DISTINCT` because `JOIN` is not supported in the current implementation? I think it would be more readable to have a `getRandomExcept` method in the `Mutator` to select all enum values except those that are not supported.

I guess you could also just extract the switch statement to a new method in `CertOracleBase` then?